### ISSUE TITLE
Incorrectly determined type of the nested conditional statement

### ISSUE THREAD

**Issue by kdmult** [vlang/v#4846]
**V version:** 0.1.27 2618b4f
**OS:** Windows 10

**What did you do?**
```
V:\>v nested_if_2.v
fn main() {
    x := 2
    z := if x > 2 { 56 } else { if x > 3 {12} else {34} }
    println(z)
}
```

**What did you expect to see?**
Successful compilation
 
**What did you see instead?**
```
V:\nested_if_2.v:5:10: error: mismatched types `int` and `bool`
    3 | fn main() {
    4 |     x := 2
    5 |     z := if x > 2 { 56 } else { if x > 3 {12} else {34} }
      |          ~~
    6 |     println(z)
    7 | }
```
It happens when nested `if` is located in the `else` part.

**Comment by krischerven** [comment]
I'm not sure if this is supposed to be valid syntax. You can do this and it compiles fine:

```v
z := if x > 2 { 56 } else if x > 3 {12} else {34}
```

**Comment by lutherwenxu** [comment]
This issue has been fixed in #4860 and could no longer be reproduced in V 0.1.27 034bf46. Closing.
